sklearn/utils/_indexing.py

```diff
@@ -520,10 +531,21 @@ def resample(*arrays, replace=True, n_samples=None, random_state=None, stratify=
         )
 
     check_consistent_length(*arrays)
-
     if stratify is None:
```

Comment:
We need to be explicit with cases that we do not implement, for instance:

```suggestion
    if sample_weight is not None and not replace:
        raise NotImplementedError(
            "Resampling with sample_weight is only implemented for replace=True."
        )
    if sample_weight is not None and stratify is not None:
        raise NotImplementedError(
            "Resampling with sample_weight is only implemented for stratify=None"
        )
    if stratify is None:
```

The docstring for the `sample_weight` parameter of `resample` needs to make those requirements explicit.

Please also add or expand an existing test in `sklearn/utils/tests/test_indexing.py` to check that those exceptions are raised when appropriate, using `with pytest.raises` blocks.